core/src/main/java/bisq/core/dao/node/full/rpc/DtoPubKeyScript.java

```diff
@@ -0,0 +1,45 @@
+package bisq.core.dao.node.full.rpc;
+
+import bisq.core.dao.state.model.blockchain.ScriptType;
+
+import bisq.common.config.Config;
+import bisq.common.util.Hex;
+
+import org.bitcoinj.script.Script;
+
+import java.util.List;
+
+import lombok.Getter;
+
+
+
+import bisq.wallets.bitcoind.rpc.responses.BitcoindScriptPubKey;
+
+@Getter
+public class DtoPubKeyScript {
+    private final String asm;
+    private final String hex;
+    private final ScriptType type;
+    private Integer reqSigs;
+    private List<String> addresses;
+
+    public DtoPubKeyScript(BitcoindScriptPubKey bitcoindScriptPubKey) {
+        this.asm = bitcoindScriptPubKey.getAsm();
+        this.hex = bitcoindScriptPubKey.getHex();
+        this.type = ScriptType.forName(bitcoindScriptPubKey.getType());
+
+        // >> addresses are not provided by bitcoin RPC from v22 onwards. <<
+        // However they are exported into the DAO classes (and therefore a component of the DAO state hash)
+        // so we must generate address from the hex script using BitcoinJ.
+        // (n.b. the DAO only ever uses/expects one address)
+        try {
+            String address = new Script(Hex.decode(hex))
+                    .getToAddress(Config.baseCurrencyNetworkParameters()).toString();
+            this.addresses = List.of(address);
+            this.reqSigs = 1;
+        } catch (Exception ex) {
+            // certain scripts e.g. OP_RETURN do not resolve to an address
```

Comment:
Shouldn't we set addresses to an empty list in that case to avoid potential NPEs?